core/src/main/java/org/apache/iceberg/io/ResolvingFileIO.java

```diff
@@ -83,6 +86,29 @@ public void deleteFile(String location) {
     io(location).deleteFile(location);
   }
 
+  @Override
+  public void deleteFiles(Iterable<String> pathsToDelete) throws BulkDeletionFailureException {
+    Map<FileIO, List<String>> pathByFileIO =
+        StreamSupport.stream(pathsToDelete.spliterator(), false)
+            .collect(Collectors.groupingBy(this::io));
+    for (Map.Entry<FileIO, List<String>> entries : pathByFileIO.entrySet()) {
+      FileIO io = entries.getKey();
+      if (io instanceof SupportsBulkOperations) {
+        ((SupportsBulkOperations) io).deleteFiles(entries.getValue());
```

Comment:
minor change, we can store entries.getValue() in a temp var as well. We only use it twice, but it will probably make the code clearer if it's "fileList" or something like that